File: examples/tasks/initialize.py
import digdag

class InstanceVariable:
    def __init__(self, arg1):
        self.arg1 = arg1

    def step1(self):
        print(self.arg1)
        print(digdag.env.params['arg1'])
        print(digdag.env.params['arg2'])
        print('self.arg1 is modified!!!')
        self.arg1 = 'var3'
        print(self.arg1)

    def step2(self):
        print('Instance variables is initialized by each tasks.')
        print(self.arg1)

File: examples/tasks/initialize.rb
class InstanceVariable
  def initialize(arg1)
    @var = arg1
  end

  def step1
    puts @var
    puts Digdag.env.params['arg1']
    puts Digdag.env.params['arg2']
    puts '@var is modified!!!'
    @var = 'var3'
    puts @var
  end

  def step2
    puts "Instance variables is initialized by each tasks."
    puts @var
  end
end
